app/jobs/square_payment_job.rb

```diff
@@ -0,0 +1,47 @@
+class SquarePaymentJob < ApplicationJob
+  class MetadataMissingError < StandardError; end
+
+  class OrderLookupError < StandardError; end
+
+  def perform(order_id:)
+    existing_adjustment = Adjustment.find_by(square_transaction_id: order_id)
+    if existing_adjustment
+      Rails.logger.info("Adjustment already exists for order #{order_id}, skipping.")
+      return
+    end
```

Comment:
We might need a better lock here in case a message is delivered more than once close enough together that it's getting processed by two workers at the same time. [The docs say](https://developer.squareup.com/docs/webhooks/overview#requirements-and-limitations):

> **Webhooks can be sent more than once.** You can bypass the processing of repeated notifications using the idempotency value included as the event_id field in the body of each event notification.